proto/src/lib.rs

```diff
@@ -20,7 +20,17 @@ mod tendermint;
 pub use tendermint::*;
 
 mod error;
-use anomaly::BoxError;
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
+#[cfg(not(feature = "std"))]
+pub use alloc::boxed::Box;
+
+#[cfg(feature = "std")]
+pub use std::boxed::Box;
```

Comment:
I think we can probably don't use feature flags here and use `alloc` crate directly. When compiled with `std` it should automatically link with the one provided by `std`.

Reply:
I have reset this change.